src/Utils/Path.idr

```diff
@@ -1,541 +1,1081 @@
-    let MkPath baseVol baseRoot baseBody _ = parse base
-    let MkPath targetVol targetRoot targetBody targetTrialSep = parse target
-    if baseVol == targetVol && baseRoot == targetRoot then Just () else Nothing
-    body <- dropBody baseBody targetBody
-    pure $ show $ MkPath Nothing False body targetTrialSep
-  where
-    dropBody : (base : List Body) -> (target : List Body) -> Maybe (List Body)
-    dropBody [] ys = Just ys
-    dropBody xs [] = Nothing
-    dropBody (x::xs) (y::ys) = if x == y then dropBody xs ys else Nothing
-
-||| Returns the last body of the path.
-|||
-||| If the last body is a file, this is the file name;
-||| if it's a directory, this is the directory name;
-||| if it's ".", it recurses on the previous body;
-||| if it's "..", returns Nothing.
-export
-fileName : String -> Maybe String
-fileName  = fileName' . parse
-
-||| Extracts the file name in the path without extension.
-|||
-||| The stem is:
-|||
-||| - Nothing, if there is no file name;
-||| - The entire file name if there is no embedded ".";
-||| - The entire file name if the file name begins with a "." and has no other ".";
-||| - Otherwise, the portion of the file name before the last ".".
-export
-fileStem : String -> Maybe String
-fileStem path = pure $ fst $ splitFileName !(fileName path)
-
-||| Extracts the extension of the file name in the path.
-|||
-||| The extension is:
-|||
-||| - Nothing, if there is no file name;
-||| - Nothing, if there is no embedded ".";
-||| - Nothing, if the file name begins with a "." and has no other ".";
-||| - Otherwise, the portion of the file name after the last ".".
-export
-extension : String -> Maybe String
-extension path = pure $ snd $ splitFileName !(fileName path)
-
-||| Updates the file name in the path.
-|||
-||| If there is no file name, it appends the name to the path;
-||| otherwise, it appends the name to the parent of the path.
-export
-setFileName : (name : String) -> String -> String
-setFileName name path = show $ setFileName' name (parse path)
-
-||| Appends a extension to the path.
-|||
-||| If there is no file name, the path will not change;
-||| if the path has no extension, the extension will be appended;
-||| if the given extension is empty, the extension of the path will be dropped;
-||| otherwise, the extension will be replaced.
-|||
-||| ```idris example
-||| "/tmp/Foo" <.> "idr" == "/tmp/Foo.idr"
-||| ```
-export
-(<.>) : String -> (extension : String) -> String
-(<.>) path ext =
-  let
-    path' = parse path
-    ext = pack $ dropWhile (\char => char == '.' || isSpace char) (unpack ext)
-    ext = if ext == "" then "" else "." ++ ext
-  in
-    case fileName' path' of
-      Just name =>
-        let (stem, _) = splitFileName name in
-          show $ setFileName' (stem ++ ext) path'
-      Nothing => path
-
-||| Drops the extension of the path.
-export
-dropExtension : String -> String
-dropExtension path = path <.> ""
+module Utils.Path
+
+
+
+import Data.List
+
+import Data.Maybe
```

Comment:
This change seems to be accidental.

Reply:
Oh dear, haha. I'll look into it.

Reply:
Seems like src/Utils/Path.idr contained Windows-style CRLF linebreak sequences and fix_whitespace was not yet able to handle those correctly. Both issues should now be fixed.